tests/testthat/test-sprintf_linter.R

```diff
@@ -91,3 +91,10 @@ test_that("edge cases are detected correctly", {
     linter
   )
 })
+
+test_that("piping into sprintf works", {
+  linter <- sprintf_linter()
+
+  expect_lint("x %>% sprintf(fmt = '%s')", NULL, linter)
+  expect_lint("x %>% sprintf(fmt = '%s%s')", "too few arguments", linter)
```

Comment:
Maybe also add a test for the format being piped?

`"%s" %>% sprintf("%s%s")` shouldn't lint for example.

Reply:
Not sure we'll catch the second case, but nor do we at HEAD... seems unlikely to come up often, and not easy to evaluate nested piping statically.

I'd push forward with the change as is for now.

Reply:
> `x |> sprintf("a") # should be ignored`

this was an error, fixed now

Reply:
No need to catch it but does it at least not cause errors?

Reply:
It's treated the same as `sprintf(x, "a")`, which we skip b/c it can't be analyzed statically:

https://github.com/r-lib/lintr/blob/2e141e95ac6c58751f43ecf58ab75f4fe465641c/tests/testthat/test-sprintf_linter.R#L105

Reply:
For `"%%sb" |> sprintf("%s") |> sprintf("a")`, it doesn't throw an error (it lints once, on the middle `sprintf()`), but I hesitate to put a test down as it sort-of commits us to whatever behavior is there (which is probably wrong)

Reply:
Added an `expect_no_error()` test for the latter case.

Reply:
> For `"%%sb" |> sprintf("%s") |> sprintf("a")`, it doesn't throw an error (it lints once, on the middle `sprintf()`), but I hesitate to put a test down as it sort-of commits us to whatever behavior is there (which is probably wrong)

That is actually correct linting for this example. What do you feel is wrong?

Reply:
Hmm I'm actually quite surprised it seems to be working for some simple nested cases:

```r
lint('x |> sprintf(fmt = "%s") |> sprintf(fmt = "%s%s")\n', sprintf_linter())
# <text>:1:29: warning: [sprintf_linter] too few arguments
# x |> sprintf(fmt = "%s") |> sprintf(fmt = "%s%s")
#                             ^~~~~~~~~~~~~~~~~~~~~
lint('x |> sprintf(fmt = "%s%s") |> sprintf(fmt = "%s")\n', sprintf_linter())
# <text>:1:6: warning: [sprintf_linter] too few arguments
# x |> sprintf(fmt = "%s%s") |> sprintf(fmt = "%s")
#      ^~~~~~~~~~~~~~~~~~~~~
```

Are we confident enough to commit these tests to the suite?

Reply:
Sounds good to me.
It's not that surprising to me if you recall that the output of sprintf is just another zappable argument for the next sprintf call in line unless it yields the fmt argument.